### PR TITLE
Remove redundant code from docker compose

### PR DESCRIPTION
**Pull request by rmoff**
I think these lines are not needed? Maybe Gabriel can confirm?

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,8 +3,6 @@ version: '2'
 services:
   zookeeper:
     image: confluentinc/cp-zookeeper:5.0.0
-    hostname: zookeeper
-    container_name: zookeeper
```
Comment (ybyzek) [inline docker-compose.yml]: @rmoff : what's the purpose to remove hostname and container_name?  They are useful in identify the containers, e.g. in output of `docker-compose ps`

File: docker-compose.yml [hunk 2]
```diff
@@ -15,11 +13,8 @@ services:
 
   kafka:
     image: confluentinc/cp-enterprise-kafka:5.0.0
-    hostname: kafka
-    container_name: kafka
     ports:
       - '9092:9092'
-      - '29092:29092'
```
Comment (ybyzek) [inline docker-compose.yml]: 29092 is the port that other containers use to connect to this Kafka broker

Comment (rmoff) [reply docker-compose.yml]: you don't need to expose the port that's used within the docker network, only for external connections from the host, and that's covered by the listener on 9092

File: docker-compose.yml (hunk 1 above)
Comment (rmoff) [reply docker-compose.yml]: The container will inherit the service name, which is why specifying these two in addition is redundant

Comment (ybyzek) [reply docker-compose.yml]: Running the docker-compose with your suggested changes results in this:

```bash
/tmp: docker-compose ps
        Name                     Command            State                           Ports                         
------------------------------------------------------------------------------------------------------------------
connect                 /etc/confluent/docker/run   Up      0.0.0.0:8083->8083/tcp, 9092/tcp                      
tmp_kafka_1             /etc/confluent/docker/run   Up      0.0.0.0:9092->9092/tcp                                
tmp_schema-registry_1   /etc/confluent/docker/run   Up      0.0.0.0:8081->8081/tcp                                
tmp_zookeeper_1         /etc/confluent/docker/run   Up      2181/tcp, 2888/tcp, 0.0.0.0:32181->32181/tcp, 3888/tcp
```

Notice the names inherit the directory name, which is what we do not want.  We want it to be called just `zookeeper`, etc

File: docker-compose.yml (hunk 2 above)
Comment (ybyzek) [reply docker-compose.yml]: True!  It's ok to remove 29092

File: docker-compose.yml (hunk 1 above)
Comment (rmoff) [reply docker-compose.yml]: Docker compose uses the directory name as a prefix for the label of the container. 
But, you can still reference it with just the service name: 

```
$ docker-compose restart zookeeper
Restarting kafka-connect-datagen_zookeeper_1 ... done
```

It's up to you if you want to keep these, it was just for brevity that I suggest removing them as they're not actually needed.

**Comment by rmoff** [comment]
Tested the PR here, all works fine: 

```
$ docker-compose ps
                 Name                              Command            State                           Ports
------------------------------------------------------------------------------------------------------------------------------------
kafka-connect-datagen_connect_1           /etc/confluent/docker/run   Up      0.0.0.0:8083->8083/tcp, 9092/tcp
kafka-connect-datagen_kafka_1             /etc/confluent/docker/run   Up      0.0.0.0:9092->9092/tcp
kafka-connect-datagen_schema-registry_1   /etc/confluent/docker/run   Up      0.0.0.0:8081->8081/tcp
kafka-connect-datagen_zookeeper_1         /etc/confluent/docker/run   Up      2181/tcp, 2888/tcp, 0.0.0.0:32181->32181/tcp, 3888/tcp

$ curl -X POST -H "Content-Type: application/json" --data @config/connector_pageviews.config http://localhost:8083/connectors
{"name":"datagen-pageviews","config":{"connector.class":"io.confluent.kafka.connect.datagen.DatagenConnector","kafka.topic":"pageviews","quickstart":"pageviews","key.converter":"org.apache.kafka.connect.storage.StringConverter","value.converter":"org.apache.kafka.connect.json.JsonConverter","value.converter.schemas.enable":"false","producer.interceptor.classes":"io.confluent.monitoring.clients.interceptor.MonitoringProducerInterceptor","max.interval":"100","iterations":"10000000","tasks.max":"1","name":"datagen-pageviews"},"tasks":[],"type":null}⏎                                                                               [~/g/kafka-connect-datagen] Robin@asgard02.moffatt.me  (nits|…)
$ curl -s "http://localhost:8083/connectors"| jq '.[]'| xargs -I{connector_name} curl -s "http://localhost:8083/connectors/"{connector_name}"/status"| jq -c -M '[.name,.connector.state,.tasks[].state]|join(":|:")'| column -s : -t| sed 's/\"//g'| sort
datagen-pageviews  |  RUNNING  |  RUNNING
[~/g/kafka-connect-datagen] Robin@asgard02.moffatt.me  (nits|…)
$ docker-compose exec connect kafka-console-consumer --topic pageviews --bootstrap-server kafka:29092  --property print.key=true --max-messages 5 --from-beginning
1       {"viewtime":1,"userid":"User_8","pageid":"Page_54"}
11      {"viewtime":11,"userid":"User_7","pageid":"Page_51"}
21      {"viewtime":21,"userid":"User_8","pageid":"Page_35"}
31      {"viewtime":31,"userid":"User_3","pageid":"Page_61"}
41      {"viewtime":41,"userid":"User_7","pageid":"Page_40"}
Processed a total of 5 messages
```

**Comment by ybyzek** [comment]
@rmoff : the alternative is to remove `docker-compose.yml` completely from the repo and just point users to https://github.com/confluentinc/cp-docker-images/blob/5.0.0-post/examples/cp-all-in-one/docker-compose.yml as an example of how to use it with `docker-compose.yml`.  The benefit of that approach is we don't have to maintain two copies.  Thoughts?

**Comment by rmoff** [comment]
I actually favour retaining the docker-compose here, as it's more concise that the all-in-one (i.e. easier to understand for someone just looking at datagen), and more lightweight for spinning up to see datagen in action. 
No harm in maybe pointing people to all-in-one as well with a note for them to see it for a full stack example?

**Comment by ybyzek** [comment]
@rmoff : I agree with your point.  Let's leave it here

**Comment by rmoff** [comment]
@ybyzek just come across this - can you re-review please? I think I addressed all your requests so far.

File: docker-compose.yml (hunk 1 above)
Comment (ybyzek) [reply docker-compose.yml]: @rmoff I'm usually in favor of brevity as well, but in this case the brevity hurts the naming of the container which should be very clear.  IMO, let's leave `hostname` and `container_name` in the docker-compose file

**Pull request closed without merging**